# crAssphage

Sequencing and analysis of crAssphage regions from around the globe and from all the metagenomes we can find.

# Paper describing this work

You can read our paper describing this work at:
[![ReadCube](images/nm.png)](https://rdcu.be/bJgsd)
[![EuropeanPMC](images/ieuropeanPMC.png)](https://europepmc.org/abstract/med/31285584)

Edwards RA, Vega AA, Norman HM, Ohaeri M, Levi K, Dinsdale EA, Cinek O, Aziz RK, McNair K, Barr JJ, Bibby K, Brouns SJJ, Cazares A, de Jonge PA, Desnues C, Díaz Muñoz SL, Fineran PC, Kurilshikov A, Lavigne R, Mazankova K, McCarthy DT, Nobrega FL, Reyes Muñoz A, Tapia G, Trefault N, Tyakht AV, Vinuesa P, Wagemans J, Zhernakova A, Aarestrup FM, Ahmadov G, Alassaf A, Anton J, Asangba A, Billings EK, Cantu VA, Carlton JM, Cazares D, Cho G-S, Condeff T, Cortés P, Cranfield M, Cuevas DA, De la Iglesia R, Decewicz P, Doane MP, Dominy NJ, Dziewit L, Elwasila BM, Eren AM, Franz C, Fu J, Garcia-Aljaro C, Ghedin E, Gulino KM, Haggerty JM, Head SR, Hendriksen RS, Hill C, Hyöty H, Ilina EN, Irwin MT, Jeffries TC, Jofre J, Junge RE, Kelley ST, Khan Mirzaei M, Kowalewski M, Kumaresan D, Leigh SR, Lipson D, Lisitsyna ES, Llagostera M, Maritz JM, Marr LC, McCann A, Molshanski-Mor S, Monteiro S, Moreira-Grez B, Morris M, Mugisha L, Muniesa M, Neve H, Nguyen N-P, Nigro OD, Nilsson AS, O’Connell T, Odeh R, Oliver A, Piuri M, Prussin AJ Ii, Qimron U, Quan Z-X, Rainetova P, Ramírez-Rojas A, Raya R, Reasor K, Rice GAO, Rossi A, Santos R, Shimashita J, Stachler EN, Stene LC, Strain R, Stumpf R, Torres PJ, Twaddle A, Ugochi Ibekwe M, Villagra N, Wandro S, White B, Whiteley A, Whiteson KL, Wijmenga C, Zambrano MM, Zschach H, Dutilh BE. 2019. Global phylogeography and ancient evolution of the widespread human gut virus crAssphage. Nat Microbiol 4:1727–1736.

This is a repository of DNA sequences, analysis, and other information for the global crAssphage project being 
developed by Bas Dutilh and Rob Edwards.

Together, we have developed this site as a common resource for everyone to add sequences, get sequences from, and build 
alignments and analyses. If you want to become a collaborator and add data to the repository, please contact Rob 
Edwards. If you want to just take the data and analyze it, you can clone this whole data set using `git clone 
https://github.com/linsalrob/crAssphage.git`. [Make sure you read the note below about Git_LFS before you do that.]

All of the data here is provided under the MIT License. Basically, you can do whatever you want as long as you 
include the original copyright and license notice in any copy of the software, source code, data, or other parts
of the repository. However, if you use this data or analysis please let Rob know. 

Please be aware that Rob and our collaborators are working on a manuscript describing this data, and so please do
not use this data as part of a manuscript without asking Rob. We will almost certainly say *yes*, unless you are
writing exactly the same manuscript that we are writing, and in which case we'll invite you to join our team!

# crAssphage sequences

## Genbank Sequence

The original sequence that was published in Genbank (but has [since been deleted](http://www.ncbi.nlm.nih.gov/nuccore/JQ995537)) is available as [JQ995537.gbk](JQ995537.gbk). It should be available in [RefSeq](http://www.ncbi.nlm.nih.gov/nuccore/674660337) although that comes and goes sometimes.

## Global Survey

Together with our [amazing collaborators](https://github.com/linsalrob/crAssphage/blob/master/COLLABORATORS.md) we have been sampling sites all over the world for crAssphage. We are still looking for more collaborators to provide sequences. Some sites have reported not finding any crAssphage, but not many!

## Local Survey

At a couple of locations we have sampled the same sites more than once. This data tells an interesting story.

## Metagenomes

We have used the awesome Hansel and Gretel to extract haplotypes from metagenomes that contain crAssphage.

## Volunteer Studies

Our awesome volunteers have provided samples time and again for us to test to see if they have crAssphage.

## Software for the analysis

We have included a list of all the software we used to analyze these sequences in [requirements.txt](requirements.txt).

## Git LFS

We use Git Large File Storage to store some of the large files. You will need to [install Git LFS](https://git-lfs.github.com/) for that to work properly. Its easy to do, and you only need to do it once.

# Questions or Comments? Want to be a collaborator?

Contact [Rob Edwards](http://edwards.sdsu.edu/research) and let him know.

